src/js/actions/index.js

```diff
@@ -0,0 +1,206 @@
+import types from '../constants/action-types';
+
+// Call action dispatch to set the filter string of search
+// Params:
+//  filter (string) = filter searchword
+export function setDogFilter(filterValue) {
+    return {
+        type: types.SET_DOG_FILTER,
+        filterValue
+    };
+}
+
+// Look for dogList in localStorage, if not found fetches list through API,
+// when dogList is retrieved calls function to dispatch action to set list in store
+export function getDogs() {
+    const dogsFromLocalStorage = getFromLocalStorage('dogList');
+    if (dogsFromLocalStorage) {
+        return setDogList(dogsFromLocalStorage);
+    }
+    return fetchDogs();
+}
+
+// Look key from local storage and return parsed object
+// Params:
+//  key (string) = key of param to look on local storage
+//  parse (bool) = if value retirved must be parsed before returning [default = true]
+// Returns: Object of what was found in local storage
+export function getFromLocalStorage(key, parse = true) {
+    const dogs = localStorage.getItem(key)
+        ? parse
+            ? JSON.parse(localStorage.getItem(key))
+            : localStorage.getItem(key)
+        : false;
+    return dogs;
+}
+
+// Fetch dog list from API and format it to an array of Objects,
+// calling the actions to set dogList or error corresponding to the fetch response
+// Params:
+//  url(string) = url to api [default = all dogs breed url]
+export function fetchDogs(url = 'https://dog.ceo/api/breeds/list/all') {
+    return dispatch => {
+        return fetch(url)
```

Comment:
You're using return inside an arrow function. This can be omitted. 
```suggestion
        return dispatch => fetch(url)
```

Reply:
🍤